Google Scholar.js

```diff
@@ -9,61 +9,239 @@
-	} else if(url.indexOf('/citations?') != -1) {
+		return "case";
+	} else if (url.indexOf('/citations?') != -1) {
+		if (getProfileResults(doc, true)) {
+			return "multiple";
+		}
+		
 		//individual saved citation
-		var link = ZU.xpathText(doc, '//a[@class="gsc_title_link"]/@href');
+		var link = ZU.xpathText(doc, '//a[@class="gsc_vcd_title_link"]/@href');
 		if(!link) return;
 		
-		if(link.indexOf('/patents?') != -1) {
-			return 'patent';
-		} else if(link.indexOf('/scholar_case?') != -1) {
+		
+		if (link.indexOf('/patents?') != -1) {
+			return 'patent';// TODO no example found
+		} else if (link.indexOf('/scholar_case?') != -1) {
 			return 'case';
 		} else {
 			//Can't distinguish book from journalArticle
 			//Both have "Journal" fields
 			return 'journalArticle';
 		}
-	} else if( getViableResults(doc).length ) {
+	} else if (getSearchResults(doc, true)) {
 		return "multiple";
 	}
 }
 
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.gs_r[data-cid]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].dataset.cid;
+		var title = text(rows[i], '.gs_rt');
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function getProfileResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('a.gsc_a_at');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].dataset.href;
+		var title = rows[i].textContent;
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	var type = detectWeb(doc, url);
+	if (type == "multiple") {
+		if (getSearchResults(doc, true)) {
+			Zotero.selectItems(getSearchResults(doc, false), function (items) {
+				if (!items) {
+					return true;
+				}
+				var ids = [];
+				for (var i in items) {
+					ids.push(i);
+				}
+				//here it is enough to know the ids and we can call scrape directly
+				scrape(doc, ids);
+			});
+		} else if (getProfileResults(doc, true)) {
+			Zotero.selectItems(getProfileResults(doc, false), function (items) {
+				if (!items) {
+					return true;
+				}
+				var articles  = [];
+				for (var i in items) {
+					articles.push(i);
+				}
+				//here we need open these pages for continue
+				ZU.processDocuments(articles, scrape);
+			});
+		}
+	} else {
+		//e.g. https://scholar.google.de/citations?view_op=view_citation&hl=de&user=INQwsQkAAAAJ&citation_for_view=INQwsQkAAAAJ:u5HHmVD_uO8C
+		scrape(doc, url, type);
+	}
+}
+
+
+function scrape(doc, idsOrUrl, type) {
+	if (Array.isArray(idsOrUrl)) {
+		scrapeIds(doc, idsOrUrl);
+	} else {
+		if (type && type=="case") {
+			scrapeCase(doc, idsOrUrl);
+		} else if (type && type=="patent") {
+			//TODO
+		} else {
+			var related = ZU.xpathText(doc, '//a[contains(@href, "q=related:")]/@href');
```

Comment:
See my much more stable solution in the new PR:
https://github.com/zotero/translators/pull/1437/files#diff-b47307f8aac17a6460d79fb3c632ee83R405

Reply:
Yes, I am also using `data-cid`. But these lines here are needed for a single page like https://scholar.google.de/citations?view_op=view_citation&hl=de&user=INQwsQkAAAAJ&citation_for_view=INQwsQkAAAAJ:u5HHmVD_uO8C or a profile page like https://scholar.google.de/citations?user=INQwsQkAAAAJ&hl=de&oi=sra .

Reply:
I see -- but wouldn't it be more reliable to use/parse the title link for this, i.e.
href="http://books.google.com/books?hl=en&lr=&id=tP8HLETgbKcC&oi=fnd&pg=PA205&dq=info:54GDwtCuzVwJ:scholar.google.com&ots=-hJyQuF2DD&sig=xPUWsIsTJzFHmglCmQCHo4ARcFU"
You already look this up at `attr(context, 'h3 a, #gsc_vcd_title a', 'href');`
the itemID is in there, too, and that's presumably guaranteed to exist for every item, which isn't the case for related.

Reply:
Here is another example https://scholar.google.de/citations?view_op=view_citation&hl=de&user=INQwsQkAAAAJ&citation_for_view=INQwsQkAAAAJ:u-x6o8ySG0sC where the extraction from the related works, but the title link is simply http://www.springerlink.com/index/rm32474088w54378.pdf

Reply:
OK. Would it make sense to check for both to catch even more cases?

Reply:
AFAS this url parameter is only present in links to GoogleBooks.

Reply:
OK, agree that's probably not worth it then.